### PR TITLE
Allow non contiguous indexing for buses

### PR DESCRIPTION
**Pull request by frapac**

### CODE REVIEW

File: src/parse/parse_mat.jl [hunk 1]
```diff
@@ -29,44 +28,49 @@ function parse_mat(file::String; kwargs...)::Dict
...
     data = Dict{String,Array}()
-    
+
     # baseMVA
     data["baseMVA"] = [data_mat["baseMVA"]]
 
     # buses
     BUS_I, BUS_TYPE, PD, QD, GS, BS, BUS_AREA, VM, VA, BASE_KV, ZONE, VMAX, VMIN,
     LAM_P, LAM_Q, MU_VMAX, MU_VMIN = IdxSet.idx_bus()
-
```
Comment (dmaldona) [inline src/parse/parse_mat.jl:42]: Thanks for the style fixes! Right now they don't come to me automatically but I'll promise I'll get better.

File: src/parse/parse_mat.jl [hunk 2]
```diff
@@ -29,44 +28,49 @@ function parse_mat(file::String; kwargs...)::Dict
...
     BUS_I, BUS_TYPE, PD, QD, GS, BS, BUS_AREA, VM, VA, BASE_KV, ZONE, VMAX, VMIN,
     LAM_P, LAM_Q, MU_VMAX, MU_VMIN = IdxSet.idx_bus()
-    
+
     nbus = length(data_mat["bus"])
     bus_array = zeros(nbus, 17)
 
+    # Keep the correspondence between Matpower's ID and ExaPF
+    # contiguous indexing.
+    bus_id_to_indexes = Dict{Int,Int}()
```
Comment (dmaldona) [inline src/parse/parse_mat.jl:47]: Great!

File: src/parse/parse_mat.jl [hunk 3]
```diff
@@ -80,16 +84,18 @@ function mat_to_exapf(data_mat)
...
-        gen_array[i, PC2] = gen["pc2"]
-        gen_array[i, QC1MIN] = gen["qc1min"]
-        gen_array[i, QC2MIN] = gen["qc2min"]
+        # pc1, pc2, qc1min, qc2min could be not specified.
+        # Set default value to 0.
+        gen_array[i, PC1] = get(gen, "pc1", 0.0)
+        gen_array[i, PC2] = get(gen, "pc2", 0.0)
+        gen_array[i, QC1MIN] = get(gen, "qc1min", 0.0)
+        gen_array[i, QC2MIN] = get(gen, "qc2min", 0.0)
     end
```
Comment (dmaldona) [inline src/parse/parse_mat.jl:93]: Thanks for this.

File: src/parse/parse_raw.jl [hunk 4]
```diff
@@ -329,6 +329,8 @@ function raw_to_exapf(rawdata::Dict{String, Array})
     end
     data["branch"] = branch_array
 
-    return data
+    # TODO:
+    bus_id_to_indexes = Dict{Int, Int}([(i, i) for i in 1:nbus])
```
Comment (dmaldona) [inline src/parse/parse_raw.jl:333]: We'll address this after we have the RGM working for a bunch of Matpower examples.

File: src/parse/parse_mat.jl (hunk 1 above)
Comment (frapac) [reply src/parse/parse_mat.jl:42]: My own code is not following exactly the style guide, I am afraid. Maybe at some point we would like to plug a formatter to autoformat the code.

File: src/parse/parse_raw.jl (hunk 4 above)
Comment (frapac) [reply src/parse/parse_raw.jl:333]: Sounds good!

**Comment by frapac** [comment]
I think it's a good idea to move the `external2internal` dict into `PowerSystem`. I am going to update the PR accordingly!

**Pull request merged**